python/pyspark/sql/types.py

```diff
@@ -441,6 +441,12 @@ def __repr__(self) -> str:
 class TimestampType(DatetimeType, metaclass=DataTypeSingleton):
     """Timestamp (datetime.datetime) data type."""
 
+    # We need to cache the timezone info for datetime.datetime.fromtimestamp
+    # otherwise the forked process will be extremely slow to convert the timestamp.
+    # This is probably a glibc issue - the forked process will have a bad cache/lock
+    # status for the timezone info.
+    tz_info = None
```

Comment:
is it possible to make timezone an optional field in `TimestampType`? 
if it is not set, then respect the config "spark.sql.session.timeZone"
@cloud-fan @HyukjinKwon 

e.g. in pyarrow,
```py
In [29]: pa.timestamp('us', tz='America/Los_Angeles')
Out[29]: TimestampType(timestamp[us, tz=America/Los_Angeles])
```

Reply:
TIMESTAMP WITH TIMEZONE type may be added in the future

Reply:
So where is the value set from?

Reply:
if TIMESTAMP WITH TIMEZONE is supported in the future, I think the TIMEZONE should be explicitly set by users, otherwise default to "spark.sql.session.timeZone"